### DISCUSSION TITLE
Outline on rounded image renders as square on Safari.

### DISCUSSION THREAD

**Discussion by kagiorgis** [tailwindlabs/tailwindcss#7649] [Help]
I used the "rounded-full" rule on an originally square img and then used the "outline" rule on it. It works fine on Chrome, but on Safari the outline renders as a square and not as a circle around the rounded img. I know it's stupid, but it's an issue.

<img
                class="w-40 rounded-full outline outline-4 outline-white dark:outline-gray-800  -mt-20  "
                src={props.profile_image}
                alt="Profile Image"
              />

![safari issue outline](https://user-images.githubusercontent.com/82416099/155642055-133fcf5e-5855-4941-90f1-d0cc4c3b8534.png)

 - OS: BigSur v.11.2
 - Browser: Safari v.14.0.3 (16610.4.3.1.4)
 - Tailwind: v3.0.23

**Comment by simonswiss** [comment]
Hey!

Yeah, this is a browser implementation thing where Safari's `outline` styles do not follow the rounded edges of an element.

Up to not so long ago, other browsers were also displaying rectangle outlines no matter what.

On the screenshot below, the circles on the left are using `outline`, and the circles on the right are using `ring` utilities.

The browsers, from top to bottom, are:

- Chrome
- Firefox
- Safari

<img width="1911" alt="Pasted_Image_25_2_22__4_09_pm" src="https://user-images.githubusercontent.com/485747/155657764-5b5acf04-33ee-4e0d-8fcb-5afc1ac6b62c.png">

For that reason, we've implemented `ring` utilities in Tailwind CSS a while back:

https://tailwindcss.com/docs/ring-width
https://tailwindcss.com/docs/ring-color
https://tailwindcss.com/docs/ring-offset-width
https://tailwindcss.com/docs/ring-offset-color

They are using `box-shadows` and let you compose some pretty advanced styles reliably.

In your case, I suggest using `ring` utilities until Safari handles rounded outline better. 

We have actually shipped new, [composable `outline` utilities](https://tailwindcss.com/docs/outline-width) in Tailwind CSS v3, because most browsers are catching up and rendering outlines pretty well:

Soon hopefully, you'll be able to use `outline` utilities to style that sort of stuff - but until then, `ring` utilities are your friends! 👍

Moving this to our GitHub discussions because I suspect some other folks might find this helpful 👍

And just in case you need it, here's the little Tailwind Play used in the screenshots:

https://play.tailwindcss.com/4mCBkc4PmD

Hope it helps!

**Reply by kagiorgis** [reply]
Thank you for your reply :)

When I used "ring" or "border", it renders correctly rounded on both browsers (chrome and safari), but on Chrome it places itself below the background img on the z axis. Any ideas on how to fix that?
```
<img
class="rounded-full border-4 border-white dark:border-gray-800 dark:outline-gray-800 "
src={props.profile_image}
alt="Profile Profile Picture"
/>
```
<img width="233" alt="ring" src="https://user-images.githubusercontent.com/82416099/155863185-75a7d426-b06a-42eb-874e-0a13a811543a.png">

**Reply by mitar** [reply]
It would be great if this would be somehow explained in the documentation. I was just now reading v3 documentation and was wondering what is the difference between ring and outline, because both rendered the same to me in Firefox, especially because outline is demonstrated on rounded buttons? Maybe outline documentation should mention that it does not yet work well on Safari? Or documentation for ring should say that browsers are catching up and one can just use outline now?

**Reply by vanling** [reply]
It's always @simonswiss who comes to the rescue when i search for an issue I have :) Thanks
